swap/swap.go

```diff
@@ -305,16 +305,24 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 		return err
 	}
 
-	opts := bind.NewKeyedTransactor(s.owner.privateKey)
-	opts.Context = ctx
-
 	otherSwap, err := contract.InstanceAt(cheque.Contract, s.backend)
 	if err != nil {
 		return err
 	}
 
-	// cash cheque in async, otherwise this blocks here until the TX is mined
-	go defaultCashCheque(s, otherSwap, opts, cheque)
+	gasPrice, err := s.backend.SuggestGasPrice(context.TODO())
+	transactionCosts := gasPrice.Uint64() * 50000 // cashing a cheque is approximately 50000 gas
+	if err != nil {
+		return err
+	}
+	paidOut, err := otherSwap.PaidOut(nil, cheque.Beneficiary)
```

Comment:
Shouldn't we check `err` here? If `err` is not `nil`, we don't need to check the following `if` right?

Reply:
Correct